AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/core/handler/ExtraHandler.java

```diff
@@ -72,12 +89,34 @@ public void process(Element element, HasExtras holder) {
 		}
 
 		JFieldVar extraKeyStaticField = createStaticExtraField(holder, extraKey, fieldName);
-		injectExtraInComponent(element, holder, extraKeyStaticField, fieldName);
-
 		if (holder instanceof HasIntentBuilder) {
 			String docComment = getProcessingEnvironment().getElementUtils().getDocComment(element);
-			createIntentInjectionMethod(element, (HasIntentBuilder) holder, extraKeyStaticField, fieldName, docComment);
+			createIntentInjectionMethod(param, (HasIntentBuilder) holder, extraKeyStaticField, fieldName, docComment);
```

Comment:
You have to add a map in the handler.

If the element is a method parameter (or an annotated method with multiple params), whose method has multiple parameters, then you have to check if all other parameters in the map or not. If they are already in, you generate the method.
If not, you just add it to the map.

If the element is a field or a method with only one param, then you can generate the extra method.